akka-persistence-testkit/src/main/scala/akka/persistence/testkit/scaladsl/UnpersistentBehavior.scala

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2022 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.testkit.scaladsl
+
+import akka.actor.typed.Behavior
+import akka.persistence.testkit.internal.{ Unpersistent, PersistenceProbeImpl }
+
+import scala.reflect.ClassTag
+
+object UnpersistentBehavior {
+  /** Given an EventSourcedBehavior, produce a non-persistent Behavior which synchronously publishes events and snapshots
+   *  for inspection.  State is updated as in the EventSourcedBehavior, and side effects are performed synchronously.  The
+   *  resulting Behavior is, contingent on the command handling, event handling, and side effects being compatible with the
+   *  BehaviorTestKit, testable with the BehaviorTestKit.
+   *
+   *  The returned Behavior does not intrinsically depend on configuration: it therefore does not serialize and assumes an
+   *  unbounded stash for commands.
+   */
+  def fromEventSourced[Command, Event, State](
+      behavior: Behavior[Command],
+      fromStateAndOffset: Option[(State, Long)] = None): (Behavior[Command], PersistenceProbe[Event], PersistenceProbe[State]) = {
+    val eventProbe = new PersistenceProbeImpl[Event]
+    val snapshotProbe = new PersistenceProbeImpl[State]
+    val resultingBehavior =
+      Unpersistent.eventSourced(behavior, fromStateAndOffset) { (event: Event, offset: Long, tags: Set[String]) =>
+          eventProbe.persist((event, offset, tags))
+        } { (snapshot, offset) =>
+          snapshotProbe.persist((snapshot, offset, Set.empty))
+        }
+
+    (resultingBehavior, eventProbe.asScala, snapshotProbe.asScala)
+  }
+
+  def fromDurableState[Command, State](
+      behavior: Behavior[Command],
+      fromState: Option[State] = None): (Behavior[Command], PersistenceProbe[State]) = {
+    val probe = new PersistenceProbeImpl[State]
+    
+    Unpersistent.durableState(behavior, fromState) { (state, version, tag) =>
+      probe.persist((state, version, if (tag.isEmpty) Set.empty else Set(tag)))
+    } -> probe.asScala
+  }
+}
+
+trait PersistenceProbe[T] {
```

Comment:
Inspired more by the `TestInbox` than `TestProbe`... `expectPersistedType` is the one thing incorporated from the `TestProbe` API.

Could incorporate the Java API's `PersistenceEffect` in place of the tuples.